pkg/kubecost/cloudcostaggregate.go

```diff
@@ -0,0 +1,440 @@
+package kubecost
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/opencost/opencost/pkg/filter"
+	"github.com/opencost/opencost/pkg/log"
+)
+
+const (
+	CloudCostAccountProp  string = "account"
+	CloudCostProjectProp  string = "project"
+	CloudCostProviderProp string = "provider"
+	CloudCostServiceProp  string = "service"
+	CloudCostLabelProp    string = "label"
+)
+
+// CloudCostAggregateProperties unique property set for CloudCostAggregate within a window
+type CloudCostAggregateProperties struct {
+	Provider   string `json:"provider"`
+	Account    string `json:"account"`
+	Project    string `json:"project"`
+	Service    string `json:"service"`
+	LabelValue string `json:"label"`
+}
+
+func (ccap CloudCostAggregateProperties) Equal(that CloudCostAggregateProperties) bool {
+	return ccap.Provider == that.Provider &&
+		ccap.Account == that.Account &&
+		ccap.Project == that.Project &&
+		ccap.Service == that.Service &&
+		ccap.LabelValue == that.LabelValue
+}
+
+func (ccap CloudCostAggregateProperties) Key(prop string) string {
+	if prop == "" {
+		return fmt.Sprintf("%s/%s/%s/%s/%s", ccap.Provider, ccap.Account, ccap.Project, ccap.Service, ccap.LabelValue)
+	}
+
+	switch prop {
+	case CloudCostProviderProp:
+		return ccap.Provider
+	case CloudCostAccountProp:
+		return ccap.Account
+	case CloudCostProjectProp:
+		return ccap.Project
+	case CloudCostServiceProp:
+		return ccap.Service
+	case CloudCostLabelProp:
+		// TODO:cloudcost will we need the LabelName here?... can move it down from the Set
+		return ccap.LabelValue
+	}
+
+	// TODO:cloudcost error?
+	return ""
```

Comment:
Have these TODOs been resolved?

Reply:
I have them resolved in a branch. I'll merge my work ASAP and those will go away.